app/controllers/concerns/foreman/controller/parameters/trend.rb

```diff
@@ -15,4 +15,14 @@ def trend_params_filter
   def trend_params
     self.class.trend_params_filter.filter_params(params, parameter_filter_context)
   end
+
+  def create_trend
```

Comment:
This is nicer than code duplication we had, but parameter handling is not exactly the right place for this code.

I think it would be better to split this code to multiple places:
Trend creation should be dealt in factory method inside common model class that we have - `Trend`:

``` ruby
class Trend < ApplicationRecord
  def self.build_trend(trend_params = {})
    trend_params[:trendable_type] == 'FactName' ? FactTrend.new(trend_params) : ForemanTrend.new(trend_params)
  end
end
```

The rest of the method should be in a concern that would be consumed by both API and UI controllers:

``` ruby
# new file: app/controllers/concerns/foreman/controller/trends.rb
module Foreman::Controller::Trends
  def create_trend
    @trend = Trend.build_trend(trend_params)
    if @trend.save
      process_success
    else
      process_resource_error
    end
  end
end
```
Now both controllers can use this newly created concern:
``` ruby
class TrendsController < V2::BaseController
  include Foreman::Controller::Parameters::Trend
  include Foreman::Controller::Trends
  # ...

  api :POST, "/trends/", N_("Create a trend counter")
  def create
     create_trend
  end
end

# in the UI controller
class TrendsController < ApplicationController
  include Foreman::Controller::Parameters::Trend
  include Foreman::Controller::Trends
  
  #...

  def create
     create_trend
  end
end
```